command/build.go

```diff
@@ -303,6 +305,25 @@ Options:
 	return strings.TrimSpace(helpText)
 }
 
-func (BuildCommand) Synopsis() string {
+func (*BuildCommand) Synopsis() string {
 	return "build image(s) from template"
 }
+
+func (*BuildCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (*BuildCommand) AutocompleteFlags() complete.Flags {
+	return complete.Flags{
+		"-color":            complete.PredictNothing,
```

Comment:
there's no way to gather these programatically? I don't like seeing a list of flags like this

Reply:
It may be possible, but I followed Vault's lead on the implementation, e.g. https://github.com/hashicorp/vault/blob/master/command/init.go#L393.

Iterating the FlagSet in this method would require ensuring all the command's state is setup before `Autocomplete*` methods are called, where the static implementation is safe to call any time.

This also allows us to editorialize, and not autocomplete deprecated flags, e.g. the `-message` flag for `packer push`.